src/main/java/org/openjdk/jextract/impl/DuplicateFilter.java

```diff
@@ -109,6 +109,15 @@ public Void visitVariable(Declaration.Variable tree, Void ignored) {
         return null;
     }
 
+    @java.lang.Override
+    public Void visitScoped(Declaration.Scoped d, Void aVoid) {
+        if (Utils.isEnum(d)) {
+            // enum constants might clash with macro constants with same name
+            d.members().forEach(m -> m.accept(this, null));
```

Comment:
Note: this will probably prefer the macro to the enum constant. While this is not ideal, this is also what happened in the old code - perhaps we can address that issue separately (it will only affect the javadoc being generated).